### DISCUSSION TITLE
Should alter_column error if passed a sa.Column object rather than a string

### DISCUSSION THREAD

**Discussion by cc-stjm** [sqlalchemy/alembic#1467] [Ideas]
I've just tracked down a bug where we were trying to alter the server_default of a column, and it wasn't applying.

The first version of the script attempted to mirror the "add_column" syntax, so for add_column we have:

```
op.add_column("table_name", sqlalchemy.Column("column_name", sa.Boolean(), server_default=False))
```
The (incorrect) syntax we were using for alter column was

```
op.alter_column("table_name", sqlalchemy.Column("column_name", sa.Boolean(), server_default=True))
```

when in fact the correct syntax is 
```
op.alter_column("table_name", "column_name", server_default=True)
```

Because `str(sqlalchemy.Column("column_name", ...)) == "column_name"`, alembic just sees this as an `alter_column` statement with no changes.

I'm wondering if some kind of error - or at least warning - here could have saved me some blushes.

Possibly a warning if the 2nd argument to alter_column is of type sa.Column? And/or if there's an alter_column statement that doesn't make any changes?

It felt like this isn't really a bug, but it's definitely a developer quality of life issue

**Comment by CaselIT** [comment]
Hi,

This is something where a type checker may be best suited for it. You will get an error there since the second arg of `alter_column` accepts only a string. Trying to check every arg in python is not very practical.
